src/adu_types/src/adu_types.c

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file adu_type.h
+ * @brief Defines common data types used throughout ADU project.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ */
+
+#include <aduc/adu_types.h>
+#include <stdlib.h>
+#include <string.h>
+
+void ADUC_ConnectionInfoDeAlloc(ADUC_ConnectionInfo* info)
+{
+    free(info->connectionString);
+    info->connectionString = NULL;
+
+    free(info->certificateString);
+    info->certificateString = NULL;
+
+    free(info->opensslEngine);
+    info->opensslEngine = NULL;
+
+    free(info->opensslPrivateKey);
+    info->opensslPrivateKey = NULL;
+
+    info->authType = ADUC_AuthType_NotSet;
+    info->connType = ADUC_ConnType_NotSet;
+}
+
+ADUC_ConnType GetConnTypeFromConnectionString(const char* connectionString)
+{
+    ADUC_ConnType result = ADUC_ConnType_NotSet;
+
+    if (strstr(connectionString, "DeviceId=") != NULL)
```

Comment:
>strstr [](start = 8, length = 6)

connectionString: `deviceid=jeff`

invalid?

Reply:
So we don't have any documentation on what a valid IoT connection string is so I had to resort to testing different cases and seeing if we could actually make a connection with anything other than "DeviceId" or "ModuleId" pascal casing and it turns out it is an enforced thing!

So this isn't a valid connection string. I'll add the note in this function but I don't know where else to look for "proper connection string structure" that isn't this one random function. 



---
In reply to: [591974012](https://github.com/Azure/iot-hub-device-update/pull/12#discussion_r591974012) [](ancestors = 591974012)